test/unit/deadline_job_attachments/bucket_sweeper/test_sweeper_processor.py

```diff
@@ -0,0 +1,240 @@
+
+    def test_upload_tag_manifest_s3_error(
+        self, processor: SweeperProcessor, mock_clients: Dict[str, Mock]
+    ):
+        """Test uploading manifest when s3 upload fails."""
+        mock_clients["s3"].upload_file.side_effect = BotoCoreError()
+
+        with pytest.raises(JobAttachmentS3BotoCoreError):
+            processor._upload_tag_manifest("test.csv", "test_key")
+
+    def test_get_manifest_etag_value_error(
+        self, processor: SweeperProcessor, mock_clients: Dict[str, Mock]
+    ):
+        """Test _get_manifest_etag method."""
+        mock_clients["s3"].head_object.return_value = {"ETag": None}
+
+        with pytest.raises(ValueError):
+            processor._get_manifest_etag("test_key")
+
+    def test_get_manifest_etag(
+        self, processor: SweeperProcessor, mock_clients: Dict[str, Mock]
+    ):
+        """Test _get_manifest_etag method."""
+        mock_clients["s3"].head_object.return_value = {"ETag": "test-etag"}
+
+        etag: str = processor._get_manifest_etag("test_key")
+        assert etag == "test-etag"
+
+        mock_clients["s3"].head_object.assert_called_once_with(
+            Bucket="test-bucket", Key="test_key"
+        )
+
+    def test_get_manifest_etag_botocore_error(
+        self, processor: SweeperProcessor, mock_clients: Dict[str, Mock]
+    ):
+        """Test _get_manifest_etag when head_object call fails."""
+        mock_clients["s3"].head_object.side_effect = BotoCoreError()
+
+        with pytest.raises(JobAttachmentS3BotoCoreError):
+            processor._get_manifest_etag("test_key")
+
+    def test_create_manifest_config(self, processor: SweeperProcessor):
+        """Test _create_manifest_config method."""
+        config: Dict[str, Any] = processor._create_manifest_config(
+            "test_key", "test-etag"
+        )
+
+        assert config == {
+            "Spec": {
+                "Format": "S3BatchOperations_CSV_20180820",
+                "Fields": ["Bucket", "Key"],
+            },
+            "Location": {
+                "ObjectArn": "arn:aws:s3:::test-bucket/test_key",
+                "ETag": "test-etag",
+            },
+        }
+
+    def test_create_delete_tagging_operation(self, processor: SweeperProcessor):
+        """Test _create_delete_tagging_operation method."""
+        operation: Dict[str, Any] = processor._create_delete_tagging_operation()
+
+        assert operation == {
+            "S3PutObjectTagging": {
+                "TagSet": [
+                    {"Key": "delete", "Value": "True"},
+                ]
+            }
+        }
+
+    def test_submit_tagging_batch_job_error(
+        self, processor: SweeperProcessor, mock_clients: Dict[str, Mock]
+    ):
+        """Test _submit_tagging_batch_job when job creation fails."""
+        mock_clients["s3_control"].create_job.side_effect = Exception("Mocked error")
+
+        with pytest.raises(SweeperProcessorError):
+            processor._submit_tagging_batch_job({}, {})
```

Comment:
```suggestion
        mock_clients["s3_control"].create_job.side_effect = Exception("Mocked error")

        with pytest.raises(SweeperProcessorError) as raised_err:
            processor._submit_tagging_batch_job({}, {})

        assert "Mocked error" in str(raised_err.value)
```

Let's make sure the error is caused by the error side effect we setup